### PR TITLE
CVSL-1505 Adding CRD and Variation subtypes

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/AdditionalCondition.kt [hunk 1]
```diff
@@ -44,4 +44,21 @@ data class AdditionalCondition(
   @Fetch(FetchMode.SUBSELECT)
   @OrderBy("id")
   val additionalConditionUploadSummary: List<AdditionalConditionUploadSummary> = emptyList(),
-)
+) {
+
```
Comment (gurps-singh) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/AdditionalCondition.kt:48]: Is this IntelliJ generated?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/Licence.kt [hunk 2]
```diff
@@ -254,260 +215,16 @@ class Licence(
...
-      "additionalConditions=$additionalConditions, " +
-      "bespokeConditions=$bespokeConditions, " +
-      "responsibleCom=$responsibleCom, " +
-      "submittedBy=$submittedBy, " +
-      "createdBy=$createdBy, " +
-      "variationOfId=$variationOfId, " +
-      "versionOfId=$versionOfId, " +
-      "licenceVersion=$licenceVersion" +
-      ")"
+    return Objects.hash(id)
```
Comment (gurps-singh) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/entity/Licence.kt:228]: Take it this was what you were talking about re the hash and equals using the id only?

**Review by gurps-singh — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt [hunk 3]
```diff
@@ -817,18 +819,14 @@ class LicenceService(
     return inFlight.isNotEmpty()
   }
 
-  private fun copyLicenceAndConditions(kind: LicenceKind, licence: EntityLicence): EntityLicence {
+  private fun copyConditions(kind: LicenceKind, licence: EntityLicence, licenceCopy: EntityLicence): EntityLicence {
```
Comment (SebNegahbanMoJ) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt:822]: Non-blocking, perhaps a slight misnomer as it's also setting the status and policy version?

**Review by SebNegahbanMoJ — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt (hunk 3 above)
Comment (andrewrlee) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceService.kt:822]: yes, it does a load of stuff really. Will update in a future PR

**Pull request merged**